File: q1-q99/q10-q19/q17/q17.py
from queue import Queue
class Solution:
    def letterCombinations(self, digits: str) -> List[str]:
        solution = []
        if not len(digits):
            return solution
        phone = [ " ", " ", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"]
        q = Queue()
        q.put("")
        for char in digits:
            index = int (char) - int("0")
            print(index)
            i = q.qsize()
            while i != 0:
                i -= 1
                top = q.get()
                for phone_char in phone[index]:
                    q.put(top+phone_char)
        while not q.empty():
            solution.append(q.get())
        return solution
                    
        

File: Sequence/q1-q99/q10-q19/q17/q17.cpp
#include <vector>
class Solution {
public:
    vector<string> letterCombinations(string digits) {
        std::vector<string> solution;
        if(!digits.size())
            return solution;
        std::vector<string> phone{ " ", " ", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz" };
        queue<string> q;
        q.push("");
                                  
        for (char digit : digits){
            int index = digit - '0';
            int i = q.size();
            while(i--){
                string top = q.front();
                q.pop();
                for(char phone_char : phone[index]){
                    q.push(top + phone_char);
                }
            }
        }
        while(!q.empty()){
            string top = q.front();
            q.pop();
            solution.push_back(top);
        }
        return solution;
    }

};
